### PR TITLE
n2v easyconfig

### PR DESCRIPTION
**Pull request by afraz-ahmed**
For INC1032175

* [x] Assigned to reviewer

n2v-0.1.11-foss-2019b-Python-3.7.4.eb

* [ ] EL7-cascadelake
* [ ] EL7-haswell
* [ ] Ubuntu16 VM

n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb

* [ ] EL7-power9
* [ ] EL7-haswell

### CODE REVIEW

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+sanity_pip_check = True
+use_pip = True
+
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+}
+
+exts_list = [
+    ('pyaml', '20.4.0', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb:31]: Does it definitely want this? I.e. that wasn't a typo for PyYAML?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+}
+
+exts_list = [
+    ('Pillow', '7.1.2', {
+        'modulename': 'PIL',
+        'checksums': ['a0b49960110bc6ff5fead46013bcb8825d101026d466f3a4de3476defe0fb0dd'],
+    }),
+    ('PyYAML', '5.3.1', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:33]: Should be a dependency

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+sanity_pip_check = True
+use_pip = True
+
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+}
+
+exts_list = [
+    ('Pillow', '7.1.2', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:29]: Should be a dependency, like the foss one. Different version though...?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    ('imagecodecs-lite', '2019.12.3', {
+        'checksums': ['95d18aa13ceb1b18a6109433b42d054e13b9a295cba96c08ab719f864f589d68'],
+    }),
+    ('tifffile', '2019.7.26.2', {
+        'checksums': ['2abb91c3a23a61593c5635ac1a19f67e732b46291c305fcee0eeaad41181a13f'],
+    }),
+    ('csbdeep', '0.4.1', {
+        'checksums': ['ad2cf81389b8c15eb2f38a524dc011a5331dfb91e7d3053fed0c6c808a7bdf89'],
+    }),
+    ('tqdm', '4.46.0', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:52]: Would https://bear-apps.bham.ac.uk/applications/tqdm/4.41.1-GCCcore-8.3.0/ suffice?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    ('imagecodecs-lite', '2019.12.3', {
+        'checksums': ['95d18aa13ceb1b18a6109433b42d054e13b9a295cba96c08ab719f864f589d68'],
+    }),
+    ('tifffile', '2019.7.26.2', {
+        'checksums': ['2abb91c3a23a61593c5635ac1a19f67e732b46291c305fcee0eeaad41181a13f'],
+    }),
+    ('csbdeep', '0.4.1', {
+        'checksums': ['ad2cf81389b8c15eb2f38a524dc011a5331dfb91e7d3053fed0c6c808a7bdf89'],
+    }),
+    ('tqdm', '4.46.0', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb:46]: Would https://bear-apps.bham.ac.uk/applications/tqdm/4.41.1-GCCcore-8.3.0/ suffice?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb (hunk 2 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:33]: I put it like this as PyYAML is not available on fosscuda/2019b. Should I build PyYAML on fosscuda-2019b first?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb (hunk 3 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:29]: Again, it hasn't been built on fosscuda. I included the latest version in the extensions list.

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb (hunk 1 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb]: turns out it was a typo!

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb (hunk 4 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb]: yes

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb (hunk 5 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb]: yes

**Comment by afraz-ahmed** [comment]
> See comments

I didn't know GCCcore-8.3.0 apps were accessible from fosscuda-2019b....

And "pyaml" was indeed a mistake!

**Comment by afraz-ahmed** [comment]
Note: tifffile is also required for PPN2V (the other installation request in this ticket).

Should I install tifffile separately, and then use it as a dependency for n2v (I can do this tomorrow), rather than an extension?

Or is it ok to use it as an extension again for PPN2V?

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+exts_list = [
+    ('Keras', '2.2.5', {
+        'checksums': ['0fb448b95643a708d25d2394183a2f3a84eefb55fb64917152a46826990113ea'],
+    }),
+    ('imagecodecs-lite', '2019.12.3', {
+        'checksums': ['95d18aa13ceb1b18a6109433b42d054e13b9a295cba96c08ab719f864f589d68'],
+    }),
+    ('tifffile', '2019.7.26.2', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:38]: You can use your new tifffile dep now

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+exts_list = [
+    ('Keras', '2.2.5', {
+        'checksums': ['0fb448b95643a708d25d2394183a2f3a84eefb55fb64917152a46826990113ea'],
+    }),
+    ('imagecodecs-lite', '2019.12.3', {
+        'checksums': ['95d18aa13ceb1b18a6109433b42d054e13b9a295cba96c08ab719f864f589d68'],
+    }),
+    ('tifffile', '2019.7.26.2', {
```
Comment (edmondac) [inline easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb:38]: You can use your new tifffile dep now

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb (hunk 6 above)
Comment (edmondac) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb:38]: You also won't need imagecodecs-lite then too

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb (hunk 7 above)
Comment (edmondac) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-foss-2019b-Python-3.7.4.eb:38]: You also won't need imagecodecs-lite then too

File: easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb (hunk 6 above)
Comment (afraz-ahmed) [reply easybuild/easyconfigs/n/n2v/n2v-0.1.11-fosscuda-2019b-Python-3.7.4.eb]: Forgot to add the modified n2v easyconfigs to the commit.

**Comment by afraz-ahmed** [comment]
> Good progress - just a couple of comments now

Forgot to add the modified n2v easyconfigs to the commit. Thanks.

**Pull request merged**